src/components/DropDownButton/index.tsx

```diff
@@ -230,25 +265,57 @@ class BaseDropDownButton extends React.Component<
 						dropUp={dropUp}
 						alignRight={alignRight}
 						onClick={this.toggle}
-						minWidth={`${this.state.minWidth}px`}
-						maxHeight={px(this.props.listMaxHeight)}
+						maxHeight={px(this.props.listMaxHeight ?? 300)}
 					>
-						{React.Children.map(children, (child, i) => {
-							if (noListFormat) {
-								return child;
-							}
-							if (!child) {
-								return;
-							}
-							if ((child as any).type === Divider) {
-								return child;
-							}
-							return (
-								<Item border={Boolean(border && i)} key={i}>
-									{child}
-								</Item>
-							);
-						})}
+						{items
+							? items.map((itemGroup, i) => (
+									<Box key={i}>
+										{itemGroup.map(
+											(item, j) =>
+												item && (
+													<Item
+														px={3}
+														py={1}
+														onClick={
+															!item.disabled && item.onClick
+																? item.onClick
+																: stopEvent
+														}
```

Comment:
What if we just do something like :

```suggestion
{...(!item.disabled && item.onClick && { onClick: item.onClick })}
```

so we can remove the stopEvent function, which could lead to strange behaviours

Reply:
Removing the stopEvent makes it so that the dropdown closes when you click a disabled item. We want it to stay open when a disabled item is click. One thing this does is help differentiate between when clicking an action has actually done something or not.